### PR TITLE
use static metric to reduce the cost of `with_label_values` in scheduler

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

Reduce the cost of `with_label_values` in scheduler.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Benchmark

The cost of `with_label_values` under `sysbench insert` workload(256 concurrency running for 400 seconds):
with-label-values

This branch:
without-label-values

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

**Comment by siddontang** [comment]
```
thread 'raftstore::test_transfer_leader::test_server_pd_transfer_leader' panicked at 'header { error { message: "stale command" } current_term: 7 }', tests/integrations/raftstore/test_transfer_leader.rs:108:9

test raftstore::test_split_region::test_node_split_epoch_not_match_left_derive ... ok

note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

stack backtrace:

   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace

             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39

   1: std::sys_common::backtrace::_print

             at src/libstd/sys_common/backtrace.rs:71

   2: std::panicking::default_hook::{{closure}}

             at src/libstd/sys_common/backtrace.rs:59

             at src/libstd/panicking.rs:197

   3: std::panicking::default_hook

             at src/libstd/panicking.rs:211

   4: std::panicking::rust_panic_with_hook

             at src/libstd/panicking.rs:474

   5: std::panicking::continue_panic_fmt

             at src/libstd/panicking.rs:381

   6: std::panicking::begin_panic_fmt

             at src/libstd/panicking.rs:336

   7: integrations::raftstore::test_transfer_leader::test_pd_transfer_leader

             at tests/integrations/raftstore/test_transfer_leader.rs:108

   8: integrations::raftstore::test_transfer_leader::test_server_pd_transfer_leader

             at tests/integrations/raftstore/test_transfer_leader.rs:116

   9: integrations::raftstore::test_transfer_leader::test_server_pd_transfer_leader::{{closure}}

             at tests/integrations/raftstore/test_transfer_leader.rs:114

  10: core::ops::function::FnOnce::call_once

             at /rustc/e305df1846a6d985315917ae0c81b74af8b4e641/src/libcore/ops/function.rs:231

  11: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once

             at /rustc/e305df1846a6d985315917ae0c81b74af8b4e641/src/liballoc/boxed.rs:704

  12: __rust_maybe_catch_panic

             at src/libpanic_unwind/lib.rs:85

  13: test::run_test::run_test_inner::{{closure}}

             at /rustc/e305df1846a6d985315917ae0c81b74af8b4e641/src/libstd/panicking.rs:272

             at /rustc/e305df1846a6d985315917ae0c81b74af8b4e641/src/libstd/panic.rs:388

             at src/libtest/lib.rs:1468

test raftstore::test_transfer_leader::test_server_pd_transfer_leader ... FAILED

```

@nolouch CI failed

**Comment by zhangjinpeng1987** [comment]
@breeswish @overvenus PTAL

**Comment by siddontang** [comment]
PTAL @fredchenbj

**Comment by siddontang** [comment]
Ping @zhangjinpeng1987 

any performance gained?

**Comment by zhangjinpeng1987** [comment]
> any performance gained?

For the reduced cost you can see benchmark in the description. If you meant sysbench/go-ycsb cases, i will bench it.

File: src/storage/metrics.rs [hunk 1]
```diff
@@ -48,12 +92,15 @@ lazy_static! {
     .unwrap();
     pub static ref KV_COMMAND_COUNTER_VEC_STATIC: KvCommandCounterVec =
         KvCommandCounterVec::from(&KV_COMMAND_COUNTER_VEC);
-    pub static ref SCHED_STAGE_COUNTER_VEC: IntCounterVec = register_int_counter_vec!(
-        "tikv_scheduler_stage_total",
-        "Total number of commands on each stage.",
-        &["type", "stage"]
-    )
-    .unwrap();
+    pub static ref SCHED_STAGE_COUNTER_VEC: SchedStageCounterVec = {
```
Comment (fredchenbj) [inline src/storage/metrics.rs]: why the '{}' is needed here?

**Comment by zhangjinpeng1987** [comment]
I deploy a cluster with 1 pd 1 tikv 1 tidb, and use sysbench to compare this branch and master. The workload is sysbench insert with concurrency is 256.


  | static metric | master
-- | -- | --
QPS | 21172 / 20717 | 20722 / 20818
avg latency | 12.09 / 12.35 | 12.35 / 12.29
.99 latency | 22.28 / 23.1 | 23.1 / 23.1

There is trivial difference from sysbench's view.

File: src/storage/metrics.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply src/storage/metrics.rs]: Just for format reason, split the long line.

**Comment by zhangjinpeng1987** [comment]
@breeswish @siddontang PTAL again.

**Pull request merged**